### PR TITLE
Add a new scraper package to replace scraperhelper

### PR DESCRIPTION
**Pull request by bogdandrutu**
Update https://github.com/open-telemetry/opentelemetry-collector/issues/11238

No changelog since this is still in progress.

### CODE REVIEW

File: scraper/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+type: service
+github_project: open-telemetry/opentelemetry-collector
+
+status:
+  class: pkg
+  stability:
+    development: [metrics]
```
Comment (songy23) [inline scraper/metadata.yaml:7]: Are traces & logs in scope too?

File: versions.yaml [hunk 2]
```diff
@@ -77,6 +77,7 @@ module-sets:
       - go.opentelemetry.io/collector/receiver/otlpreceiver
       - go.opentelemetry.io/collector/receiver/receiverprofiles
       - go.opentelemetry.io/collector/receiver/receivertest
+      - go.opentelemetry.io/collector/scraper
```
Comment (songy23) [inline versions.yaml:80]: Please update https://github.com/open-telemetry/opentelemetry-collector/blob/main/cmd/builder/test/core.builder.yaml as well

Comment (bogdandrutu) [reply versions.yaml:80]: No need for the moment, since noboday depends on it.

File: scraper/metadata.yaml (hunk 1 above)
Comment (bogdandrutu) [reply scraper/metadata.yaml:7]: I think we will extend to logs at one point, see request https://github.com/open-telemetry/opentelemetry-collector/issues/11238#issuecomment-2457770329.

For me the first part is to move what we have (only metrics) to this new framework proposed in the issue, then we can extend to logs.

**Review by songy23 — APPROVED** [review]

**Comment by sincejune** [comment]
@bogdandrutu May I ask what would be the next step of this PR? Are we planning to remove scraperhelper/scrapercontroller as well?

**Pull request merged**